### PR TITLE
Fixed bug with sorting by title being case sensitive

### LINKED ISSUE #631: Sorting by title does not work properly
It seems to be case sensitive

### PR DESCRIPTION
**Pull request by giocalitri**
Fixes #631

### CODE REVIEW

File: search/search_indexes.py [hunk 1]
```diff
@@ -137,6 +136,15 @@ def prepare_preview_url(self, obj):  # pylint: disable=no-self-use
             run=data["run"],
         )
 
+    def prepare_titlesort(self, obj):  # pylint: disable=no-self-use
+        """Define what goes into the "titlesort" index."""
+        title = obj.title.strip()
+        # hack to handle empty titles
+        # to show up at the bottom of the sorted list
+        if not title:
+            title = 'z' * 30
```
Comment (noisecapella) [inline search/search_indexes.py]: Why not prepend a '0' to all titles and then make the title a '1' if it's empty? That guarantees it's at the bottom. We can't rely on ASCII comparisons like this, what if the title starts with a non alphanumeric character, or something unicode?

Comment (noisecapella) [inline search/search_indexes.py]: To clarify, I mean relying on 'z' having a high value relative to everything else. It's reasonable to expect '0' to be before '1' and 'a' to be before 'z'

File: search/sorting.py [hunk 2]
```diff
@@ -13,7 +13,7 @@ class LoreSortingFields(object):
     SORT_BY_NR_VIEWS = ('nr_views', 'Number of Views (desc)', '-')
     SORT_BY_NR_ATTEMPTS = ('nr_attempts', 'Number of Attempts (desc)', '-')
     SORT_BY_AVG_GRADE = ('avg_grade', 'Average Grade (desc)', '-')
-    SORT_BY_TITLE = ('title', 'Title (asc)', '')
+    SORT_BY_TITLE = ('titlesort', 'Title (asc)', '')
```
Comment (noisecapella) [inline search/sorting.py:16]: nit: It seems a bit odd to have `?sortby=titlesort` in the URL, is there some way to use `title` in the URL but use the `titlesort` field in the index? Feel free to ignore if this complicates things unnecessarily

File: search/search_indexes.py (hunk 1 above)
Comment (giocalitri) [inline search/search_indexes.py]: I agree with your solution: will do

File: search/sorting.py (hunk 2 above)
Comment (giocalitri) [inline search/sorting.py:16]: yes, but it means changing many more things:
what about a different string instead of `titlesort`? Any idea?

Comment (noisecapella) [inline search/sorting.py:16]: If we can't use `title`, `titlesort` is fine with me

**Comment by giocalitri** [comment]
Updated the `prepare_titlesort` function

**Comment by noisecapella** [comment]
:+1: after tests pass

**Pull request merged**